Use GitHub actions for testing and deployment

adamkudrna opened the pull request:
Check the workflow run in my fork: https://github.com/adamkudrna/egamebook_site/actions/runs/561249069

Build now fails on permissions for Firebase which I don't have. You can download the `site` artifact down on the page above to see if there is everything as expected.

adamkudrna commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,89 @@
+    branches:
+      # TODO change to prod
+      # - prod
+      - feature/github-actions
```
⚠️ This needs to be changed to `prod` before this PR is merged.

filiph commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,89 @@
+jobs:
+  build_and_test:
+    name: Build and test
+    runs-on: ubuntu-18.04
```
Wouldn't it be better to collapse the two workflows (`build-and-test.yml` and `deploy.yml` into one and add this here?

```suggestion
    runs-on: ubuntu-18.04
    if: github.ref == 'refs/heads/prod'
```

References:

* https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif
* https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions

filiph replied:
Actually, more like:

```suggestion
    needs: build_and_test
    if: |
      github.event_name == 'push'
      && github.ref == 'refs/heads/prod'
      && github.repository == 'filiph/egamebook_site'
    runs-on: ubuntu-18.04
```

adamkudrna replied:
I see what you mean. Let's do that. Although I prefer to keep these things separate for easier extensibility, there is a clear code duplicity which cannot be avoided (now) without moving everything into a single file (cross-workflow artefact sharing is not yet supported). I'll only get rid of test with Node 15 which is not a big deal.

filiph reviewed (approved):
LGTM with two suggestions. Thank you!

You can also prepare the PR to already use `prod`. I have created the branch and added the firebase token.

adamkudrna commented on README.md:
```diff
@@ -1,40 +1,87 @@
+## Deployment
+
+The site is automatically deployed to Firebase hosting on push to the `prod`
+branch.
```
Fixed.

adamkudrna commented:
@filiph I addressed your suggestions **and rebased my branch** (so you don't merge my fixup commits by accident).

Build result on my side: https://github.com/adamkudrna/egamebook_site/actions/runs/592435994
